nex/friends-wiiu/mark_friend_requests_as_received.go

```diff
@@ -1,16 +1,30 @@
 package nex_friends_wiiu
 
 import (
-	database_wiiu "github.com/PretendoNetwork/friends-secure/database/wiiu"
-	"github.com/PretendoNetwork/friends-secure/globals"
+	database_wiiu "github.com/PretendoNetwork/friends/database/wiiu"
+	"github.com/PretendoNetwork/friends/globals"
 	nex "github.com/PretendoNetwork/nex-go"
 	friends_wiiu "github.com/PretendoNetwork/nex-protocols-go/friends-wiiu"
 )
 
-func MarkFriendRequestsAsReceived(err error, client *nex.Client, callID uint32, ids []uint64) {
+func MarkFriendRequestsAsReceived(err error, client *nex.Client, callID uint32, ids []uint64) uint32 {
+	if err != nil {
+		globals.Logger.Error(err.Error())
+		return nex.Errors.FPD.InvalidArgument
+	}
+
+	var markErr error
 	for i := 0; i < len(ids); i++ {
 		id := ids[i]
-		database_wiiu.SetFriendRequestReceived(id)
+		err = database_wiiu.SetFriendRequestReceived(id)
+		if err != nil {
+			globals.Logger.Critical(err.Error())
+			markErr = err
+		}
```

Comment:
I'm not sold on the idea that we should continue processing data when an error occurs. Though I'm also not really sure of a better way to handle this. Thoughts?

Reply:
I had the same doubts, and I'm not sure of another way to handle it. This issue will probably appear on some database getters too, so we have to think about it.

I will bring up @hauntii for suggestions on this

Reply:
I think we should abort processing here, just because if the console gets an error (which it will no matter if we abort or not) it's just going to try again anyways, so there's no point in continuing if it's going to try and remark them anyways.

For the others that retrieve user lists in the others mentioned, I also don't think we should keep going since we don't want the user to get an incomplete list of their friends / blocked and think that they've been unfriended / their block has inexplicably disappeared.

Reply:
That makes sense, thank you!